.github/workflows/main.yml

```diff
@@ -1814,6 +2002,42 @@ jobs:
         if: failure()
         with:
           paths: "./*/*/build/test/*/results/*.xml"
+
+
+  graalvm-test:
+    name: GraalVM Test ${{ matrix.graal }}
+    needs: base-build
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        graal: [ '20.3.0', '21.1.0' ]
+    steps:
+
+      - name: Set up JDK 8
+        uses: actions/setup-java@v3
+        with:
+          java-version: 8 
+          distribution: 'zulu'
+
+      - name: Setup Xvfb
+        run: |
+          echo "DISPLAY=:99.0" >> $GITHUB_ENV
+          Xvfb :99 -screen 0 1024x768x24 > /dev/null 2>&1 &
+
+      - name: Download Build
+        uses: actions/download-artifact@v3
+        with:
+          name: build
+
+      - name: Extract
+        run: tar --zstd -xf build.tar.zst
+        
+      - name: Setup Env
+        run: echo "GRAAL_VERSION=${{ matrix.graal }}" >> $GITHUB_ENV
+
+      - name: nbbuild/travis/scripting.sh
+        run: nbbuild/travis/scripting.sh
```

Comment:
i thought about converting scripting.sh into gh yaml

this would map directly to steps. Makes it easier to see what fails (log split per step), allows to add retry per test if needed etc. (context: the job was fairly unreliable on travis)

Disadvantage is that it makes it somewhat less portable + if someone likes to run this locally it wouldn't be a single script anymore. I could leave the setup part in the script and call it, while converting the test execution to steps?

@neilcsmith-net @matthiasblaesing @jtulach what do you think?

Reply:
or: i don't touch the script, i simply replicate it in the gh yaml. Since the graalvm setup can be probably mapped to the java setup action

Reply:
I ended up fully converting the script to steps.